src/states/node.rs

```diff
@@ -1877,98 +1852,117 @@ impl Node {
             .unwrap_or_else(|| utils::calculate_relocation_dst(close_section, &dst_name));
 
         // From X -> Y; Send to close section of the relocated name
-        let request_content = MessageContent::ExpectCandidate {
+        let request_content = MessageContent::ExpectCandidate(ExpectCandidate {
             old_public_id: relocating_node_id,
             old_client_auth: Authority::Client {
                 client_id: relocating_node_id,
                 proxy_node_name: proxy_name,
             },
             message_id: message_id,
-        };
+        });
 
         let src = Authority::Section(dst_name);
         let dst = Authority::Section(relocation_dst);
         self.send_routing_message(src, dst, request_content)
     }
 
     // Received by Y; From X -> Y
-    // Context: a node is joining our section. Sends `AcceptAsCandidate` to our section. If the
-    // network is unbalanced, sends `ExpectCandidate` on to a section with a shorter prefix.
+    // Context: a node is joining our section. Vote `ExpectCandidate`.
     fn handle_expect_candidate(
         &mut self,
-        old_pub_id: PublicId,
-        old_client_auth: Authority<XorName>,
-        relocation_dst: Authority<XorName>,
-        message_id: MessageId,
+        rpc: ExpectCandidate,
+        dst_name: XorName,
     ) -> Result<(), RoutingError> {
-        self.remove_expired_peers();
+        if !self.chain.is_member() || !self.peer_mgr.is_established() {
```

Comment:
why are we needing both checks than just `chain.is_member()` here?

Reply:
On my first attempt, tests were still failing, but trying it again, it seem to be enough.
Will fix.